### PR TITLE
Adding new device for Cook Surface. This is needed for the Cooktop De…

### PR DESCRIPTION
**Pull request by ap-whirlpool**
This PR is for introducing the Cook Surface. 

This creates a new device under matter-devices.xml and updates the zap regen files.

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -2363,4 +2363,19 @@ limitations under the License.
...
+        <domain>CHIP</domain>
+        <typeName>Matter Cook Surface</typeName>
+        <class>Simple</class>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x0077</deviceId>
+        <clusters lockOthers="true">
+            <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
+            </include>
+            <include cluster="Temperature Control" client="false" server="true" clientLocked="true" serverLocked="true"></include>
+            <include cluster="Temperature Measurement" client="false" server="true" clientLocked="true" serverLocked="true">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2377]: This requires both of those clusters, no?  But the spec says at least one has to be supported, not necessarily both.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 1 above)
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Did my latest change fix this @bzbarsky-apple ? If not, what should I change this to? Or if there is another example I can relate it to.
My latest change was to use the same methodology that was used from the refrigerator device.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -2363,4 +2363,18 @@ limitations under the License.
...
+        <domain>CHIP</domain>
+        <typeName>Matter Cook Surface</typeName>
+        <class>Simple</class>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x0077</deviceId>
+        <clusters lockOthers="true">
+            <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
+            </include>
+            <include cluster="Temperature Control" client="false" server="true" clientLocked="true" serverLocked="true"></include>
+            <include cluster="Temperature Measurement" client="false" server="false" clientLocked="true" serverLocked="false"></include>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2377]: You want serverLocked="false" for both of these, no?  It should be valid to not have "Temperature Control".

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: You want serverLocked="false" for both of the optional things.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Oops forgot that one. Updated.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 1 above)
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Updated.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

**Review by fessehaeve — APPROVED** [review]

**Pull request merged**